nautobot_floor_plan/forms.py

```diff
@@ -89,3 +95,29 @@ class Meta:
             "rack_orientation",
             "tags",
         ]
+
+    def __init__(self, *args, **kwargs):
+        """Overwrite the constructor to swap x/y origins fields to CharField if letters are used for grid numbering."""
+        super().__init__(*args, **kwargs)
+        self.x_letters = False
+        self.y_letters = False
+        char_field = self.fields.pop("_char")
+
+        if fp_id := self.initial["floor_plan"] or self.data["floor_plan"]:
+            fp_obj = self.fields["floor_plan"].queryset.get(id=fp_id)
+            if fp_obj.x_axis_labels == choices.AxisLabelsChoices.LETTERS:
+                self.x_letters = True
+                self.fields["x_origin"] = char_field
+            if fp_obj.y_axis_labels == choices.AxisLabelsChoices.LETTERS:
+                self.y_letters = True
+                self.fields["y_origin"] = char_field
+
+    def clean(self):
+        """Change x/y origins back to numbers."""
+        if self.x_letters:
+            self.cleaned_data["x_origin"] = utils.column_letter_to_num(self.cleaned_data.get("x_origin"))
```

Comment:
Should we split these out to proper `clean_x_origin` and `clean_y_origin` methods?

Reply:
I started with that, but I was facing some exception because clean() seems to be called before.
Maybe changing the form field to CharField will resolve that.

Reply:
Yes, after simplifying form to use just CharFields, clean_ method for each fields just work fine.